pmm-app/src/shared/components/Form/NumericInput/NumericInput.tsx

```diff
@@ -0,0 +1,83 @@
+import React, { FC, useCallback, useRef } from 'react';
+import { cx } from 'emotion';
+import { Input as BaseInput, useTheme } from '@grafana/ui';
+import { Props as InputProps } from '@grafana/ui/components/Input/Input';
+import { InputLabel } from 'shared/components/Form';
+import { getStyles } from './NumericInput.style';
+
+interface NumericInputProps extends InputProps {
+  label?: string;
+  labelWidth?: number;
+}
+
+export const NumericInput: FC<NumericInputProps> = ({
+  className = '',
+  disabled = false,
+  label = '',
+  labelWidth,
+  ...props
+}) => {
+  const inputRef = useRef<HTMLInputElement>(null);
+  const theme = useTheme();
+  const styles = getStyles(theme);
+
+  const dispatchChangeEvent = useCallback(() => {
+    const event = new Event('change', { bubbles: true });
+
+    inputRef.current!.dispatchEvent(event);
+  }, [inputRef]);
+
+  const stepUp = () => {
+    inputRef.current!.stepUp();
+    dispatchChangeEvent();
+  };
+
+  const stepDown = () => {
+    inputRef.current!.stepDown();
+    dispatchChangeEvent();
+  };
+
+  return (
+    <span className={styles.wrapper}>
+      {label && (
+        <div>
+          <InputLabel style={labelWidth ? { width: labelWidth * 8, minWidth: 0 } : undefined}>
```

Comment:
Not sure if we want to have it as a factor of eight. Seems to be less flexible than just passing an arbitrary number.

Reply:
Yes, as we discussed: this is the approach Grafana's taking for their components, we're free to use what we want. Just notice that it will behave differently from other native Grafana components (which use multiples of 8) if we use arbitrary numbers

Reply:
Ok, let's leave 8x for the time being. I guess we need to see it practice then maybe revisit.